engine/table/src/test/java/io/deephaven/engine/table/impl/select/TestCompareUtils.java

```diff
@@ -242,6 +242,63 @@ public void testPrimitives() {
         assertEquals(0, CompareUtils.compare(1.0, 1.0f));
     }
 
+    public void testBoolean() {
+        // boolean vs boolean
+        assertTrue(CompareUtils.compare(true, false) > 0);
+        assertTrue(CompareUtils.compare(false, true) < 0);
```

Comment:
I think we should refactor this whole test class, and have a helper method that does all of these flip-flopped comparisons instead of making each test do it themselves:

```java
static void assertLt(Object lower, Object upper) {
  assertTrue(CompareUtils.compare(lower, upper) < 0);
  assertTrue(CompareUtils.compare(upper, lower) > 0);
}

// note: no assertGt, just call assertLt w/ args in order

static void assertEq(Object lower, Object upper) {
  assertEquals(0, CompareUtils.compare(lower, upper));
  assertEquals(0, CompareUtils.compare(upper, lower));
}
```